demo/ios/BackgroundService/PorcupineBackgroundServiceDemo/ViewController.swift

```diff
@@ -28,34 +29,35 @@ class ViewController: UIViewController, UITextViewDelegate {
         startButton.clipsToBounds = true
         
         textView.text = "Press the Start button and say the wake word \"Porcupine\". Try pressing the home button and saying it again."
-        
-        Sound.category = .playAndRecord
-        let keywordCallback: ((Int32) -> Void) = { keywordIndex in
-            NotificationManager.shared.sendNotification()
-            Sound.play(file: "beep.wav")
-        }
-        do {
-            self.porcupineManager = try PorcupineManager(keyword: wakeWord, onDetection: keywordCallback)
-        } catch {
-            showAlert(message: "Failed to initialize Porcupine Manager")
-        }
     }
 
     @IBAction func toggleStartButton(_ sender: UIButton) {
         if !isRecording {
-            NotificationManager.shared.requestNotificationAuthorization()
-
             do {
+                Sound.category = .playAndRecord
+                let keywordCallback: ((Int32) -> Void) = { keywordIndex in
+                    NotificationManager.shared.sendNotification()
+                    Sound.play(file: "beep.wav")
+                }
+                
+                self.porcupineManager = try PorcupineManager(accessKey: accessKey, keyword: wakeWord, onDetection: keywordCallback)
+                
+                
                 try porcupineManager.start()
-            } catch {
-                showAlert(message: "Failed to start Porcupine Manager")
+                isRecording = true
+                startButton.setTitle("STOP", for: UIControl.State.normal)
+            } catch let error as PorcupineError {
+                showAlert(message: "Failed to initialize and start Porcupine:\n\(error)")
+            } catch let error {
+                showAlert(message: "Something went wrong: \(error)")
                 return
             }
 
-            isRecording = true
-            startButton.setTitle("STOP", for: UIControl.State.normal)
+
         } else {
-            porcupineManager.stop()
+            if porcupineManager != nil {
```

Comment:
At the top we assert that PorcupineManager will have a value (with the `!`). If we ever think it won't have a value, we need to change the type to `?`

Reply:
I was noticing when the `PorcupineManager` constructor failed it was left without a value, but I guess with the changes to how we are handling error's it's okay.